src/sentry/dynamic_sampling/tasks.py

```diff
@@ -102,3 +117,90 @@ def adjust_sample_rates(
                 project_id=ds_project.id, trigger="dynamic_sampling_prioritise_project_bias"
             )
         pipeline.execute()
+
+
+@instrumented_task(
+    name="sentry.dynamic_sampling.tasks.prioritise_transactions",
+    queue="dynamicsampling",
+    default_retry_delay=5,
+    max_retries=5,
+    soft_time_limit=2 * 60 * 60,  # 2hours
+    time_limit=2 * 60 * 60 + 5,
+)  # type: ignore
+def prioritise_transactions() -> None:
+    """
+    A task that retrieves all relative transaction counts from all projects in all orgs
+    and invokes a task for rebalancing transaction sampling rates within each project
+    """
+    metrics.incr("sentry.tasks.dynamic_sampling.prioritise_transactions.start", sample_rate=1.0)
+    current_org: Optional[Organization] = None
+    current_org_enabled = False
+    with metrics.timer("sentry.tasks.dynamic_sampling.prioritise_transactions", sample_rate=1.0):
+        for orgs in get_orgs_with_project_counts(MAX_ORGS_PER_QUERY, MAX_PROJECTS_PER_QUERY):
+            # get the low and high transactions
+            # TODO can we do this in one query rather than two
+            for project_transactions in transactions_zip(
+                fetch_transactions_with_total_volumes(
+                    orgs,
+                    large_transactions=True,
+                    max_transactions=MAX_TRANSACTIONS_PER_PROJECT // 2,
+                ),
+                fetch_transactions_with_total_volumes(
+                    orgs,
+                    large_transactions=False,
+                    max_transactions=MAX_TRANSACTIONS_PER_PROJECT // 2,
+                ),
+            ):
+
+                if not current_org or current_org.id != project_transactions["org_id"]:
+                    current_org = Organization.objects.get_from_cache(
+                        id=project_transactions["org_id"]
+                    )
+                    current_org_enabled = features.has(
```

Comment:
I would be inclined to move this check before the transactions queries. Why fetch and process all the transactions if we are just going to throw them away?

Reply:
Thanks @evanh  for you input.
Yes, you are absolutely right, we have debated about checking early and getting the transactions only for the
orgs that are enabled.

The reason we decided to go for this is that we wanted to check if we are able to run the queries for everything and for now only apply the changes to a limited group of orgs (that we use for testing).

The feature is only intended to be temporary, once we are happy with the result on our test orgs we will enable this for everybody and we don't want to have the unpleasant surprise that the database cannot handle it.

If we have any doubts about the query performance we can use the same approach that we used for project and do the first query ( the query for orgs) using modulo i.e. `Condition(Function("modulo", [Column("org_id"), 100]), Op.LT, load_rate)`. This way we can ramp up db load independently from enabling the organisations for transaction boosting.

WDYT ?

Reply:
Done.